platform/backend/src/mcp-server-runtime/manager.ts

```diff
@@ -40,36 +113,47 @@ export class McpServerRuntimeManager {
   constructor() {
     this.k8sConfig = new k8s.KubeConfig();
 
-    // Load K8s config from environment or default locations
+    const hasCustomKubeconfig =
+      Boolean(kubeconfig && kubeconfig.trim().length > 0);
+
     try {
+      // If not loading from cluster AND user provided a custom kubeconfig → validate it
+      if (!loadKubeconfigFromCurrentCluster && hasCustomKubeconfig) {
+        validateKubeconfig(kubeconfig!);
```

Comment:
It's easier for the reader if the validation check is moved into `} else if (hasCustomKubeconfig) {` block.